test/test_hyperparameters.py

```diff
@@ -1838,6 +1838,13 @@ def test_categorical_choices(self):
         ):
             CategoricalHyperparameter('param', ['a', None])
 
+    def test_categorical_default(self):
+        f1 = CategoricalHyperparameter("param", ["a", "b"])
+        f2 = CategoricalHyperparameter("param", ["a", "b"], weights=[0.3, 0.6])
+        f3 = CategoricalHyperparameter("param", ["a", "b"], weights=[0.6, 0.3])
+        self.assertNotEqual(f1.default_value, f2.default_value)
+        self.assertEqual(f1.default_value, f3.default_value)
```

Comment:
I won't block on this but reading the test alone doesn't illuminate what the intended behavior is meant to be. Either a comment or to say so or a single assert per `fX` could help